radixdlt-core/radixdlt/src/test/java/com/radixdlt/statecomputer/radixengine/AbstractRadixEngineTest.java

```diff
@@ -0,0 +1,166 @@
+
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Inject;
+import com.google.inject.Scopes;
+import com.google.inject.multibindings.Multibinder;
+import com.radixdlt.SingleNodeAndPeersDeterministicNetworkModule;
+import com.radixdlt.api.core.reconstruction.BerkeleyRecoverableProcessedTxnStore;
+import com.radixdlt.application.system.FeeTable;
+import com.radixdlt.application.tokens.Amount;
+import com.radixdlt.application.validators.state.ValidatorRegisteredCopy;
+import com.radixdlt.consensus.bft.Self;
+import com.radixdlt.crypto.ECKeyPair;
+import com.radixdlt.environment.deterministic.DeterministicProcessor;
+import com.radixdlt.mempool.MempoolConfig;
+import com.radixdlt.network.p2p.P2PConfig;
+import com.radixdlt.network.p2p.RadixNodeUri;
+import com.radixdlt.network.p2p.addressbook.AddressBook;
+import com.radixdlt.network.p2p.addressbook.AddressBookPersistence;
+import com.radixdlt.networks.NetworkId;
+import com.radixdlt.properties.RuntimeProperties;
+import com.radixdlt.statecomputer.checkpoint.MockedGenesisModule;
+import com.radixdlt.statecomputer.forks.ForksModule;
+import com.radixdlt.statecomputer.forks.MainnetForkConfigsModule;
+import com.radixdlt.statecomputer.forks.RERulesConfig;
+import com.radixdlt.statecomputer.forks.RadixEngineForksLatestOnlyModule;
+import com.radixdlt.store.DatabaseLocation;
+import com.radixdlt.store.berkeley.BerkeleyAdditionalStore;
+import com.radixdlt.utils.PrivateKeys;
+import com.radixdlt.utils.UInt256;
+import java.util.Map;
+import java.util.Set;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.rules.TemporaryFolder;
+
+public abstract class AbstractRadixEngineTest {
+  private static final ECKeyPair TEST_KEY = PrivateKeys.ofNumeric(1);
+
+  @Rule public TemporaryFolder folder = new TemporaryFolder();
+  @Inject private DeterministicProcessor processor;
+
+  private final Amount totalTokenAmount = Amount.ofTokens(110);
+  private final Amount stakeAmount = Amount.ofTokens(10);
+  private final int mempoolMaxSize;
+
+  protected AbstractRadixEngineTest(int mempoolMaxSize) {
+    this.mempoolMaxSize = mempoolMaxSize;
+  }
+
+  @Before
+  public void setup() {
+    var injector =
+        Guice.createInjector(
+            MempoolConfig.asModule(mempoolMaxSize, 10),
+            new MainnetForkConfigsModule(),
+            new RadixEngineForksLatestOnlyModule(
+                RERulesConfig.testingDefault()
+                    .overrideFeeTable(
+                        FeeTable.create(
+                            Amount.ofSubunits(UInt256.ONE),
+                            Map.of(ValidatorRegisteredCopy.class, Amount.ofSubunits(UInt256.ONE))))
+                    .overrideMaxMessageLen(511)),
```

Comment:
For a class called a generic name such as `AbstractRadixEngineTest`, I find this override unexpected.

Is it worth instead having the class take some optional overrides in the constructor, to make the values of the important parameters in subclasses explicit in those classes?

Reply:
Actually, this is the same issue as one mentioned above - `RERulesConfig` has factory and `overrideXXX` methods which are used exclusively in tests. To address this issue, I've created a ticket (see above).

Reply:
My point is that, in a test base class, I don't feel like we should have unexpected/arbitrary overrides of defaults unless clearly required / specified by a subclass. Otherwise the test set-up is hidden/implicit instead of explicit, resulting in a more fragile and harder to understand test.

Reply:
I've changed the test setup to make this limit coming directly from the particular test.